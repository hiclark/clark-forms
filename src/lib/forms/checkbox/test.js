import React from 'react';

import Checkbox from './';

describe('<Checkbox />', () => {
  let wrapper;
  let checkbox;
  let label;

  beforeEach(() => {
    const props = {
      onCheckboxClick: jest.fn(),
      checked: true,
      name: 'Name',
      label: 'Label',
      index: 1,
      value: false,
    };
    wrapper = shallow(<Checkbox {...props} />);
    checkbox = wrapper.find('Field');
    label = wrapper.find('label');
  });

  describe('<Checkbox /> field renders props correctly', () => {
    test('render value prop', () => {
      expect(checkbox.prop('value')).toBeFalsy();
    });

    test('render index prop', () => {
      expect(checkbox.prop('index')).toEqual(1);
    });

    test('render name prop', () => {
      expect(checkbox.prop('name')).toMatch('Name');
    });
  });

  describe('<Checkbox /> label renders correctly', () => {
    test('render label text', () => {
      expect(
        label
          .find('checkbox__LabelText')
          .render()
          .text(),
      ).toEqual('Label');
    });

    test('render label icon', () => {
      expect(
        label
          .find('checkbox__CheckboxLabel')
          .childAt(0)
          .type(),
      ).toEqual('toggle-on.svg');
    });

    test('simulates onCheckboxClick click event', () => {
      expect(label.props().onClick).toHaveBeenCalledTimes(0);
      label.simulate('click');
      expect(label.props().onClick).toHaveBeenCalledTimes(1);
    });

    test('onCheckboxClick click event renders icon correctly', () => {
      expect(
        wrapper
          .find('label')
          .find('checkbox__CheckboxLabel')
          .childAt(0)
          .type(),
      ).toEqual('toggle-on.svg');
      wrapper.setProps({ checked: false });
      expect(
        wrapper
          .find('label')
          .find('checkbox__CheckboxLabel')
          .childAt(0)
          .type(),
      ).toEqual('toggle-off.svg');
    });
  });

  test('snapshot test <Checkbox />', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
