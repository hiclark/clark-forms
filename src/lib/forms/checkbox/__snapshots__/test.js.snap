// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> when checked is false matches the snapshot 1`] = `
<div>
  <Field
    checked={false}
    component={[Function]}
    index={1}
    name="Name"
    onCheckboxClick={[Function]}
    value={false}
  />
</div>
`;

exports[`<Checkbox /> when checked is true matches the snapshot 1`] = `
<div>
  <Field
    checked={true}
    component={[Function]}
    index={1}
    name="Name"
    onCheckboxClick={[Function]}
    value={false}
  />
</div>
`;

exports[`<Checkbox /> when variant is addSub matches the snapshot 1`] = `
<div>
  <Field
    checked={true}
    component={[Function]}
    index={1}
    name="Name"
    onCheckboxClick={[Function]}
    value={false}
    variant="addSub"
  />
</div>
`;

exports[`<Checkbox /> with label matches the snapshot 1`] = `
<div>
  <Field
    checked={true}
    component={[Function]}
    index={1}
    label="Label"
    name="Name"
    onCheckboxClick={[Function]}
    value={false}
  />
</div>
`;
